### PR TITLE
Added NoEmptyPhpdocFixer

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1688

### CODE REVIEW

**Comment by keradus** [comment]
`NoEmptyPhpdocFixer` ?

File: Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    /**  *//** a *//**  */
+
+                    /**
+                    */
+
+                    /**
+                     *
+                     */
+
+                    /** ***
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php]: This is not empty docblock. it's odd, but not empty. The Summary is `***`.

File: README.rst [hunk 2]
```diff
@@ -615,6 +615,10 @@ Choose from the list of available fixers:
                         construct to print if
                         possible.
 
+* **empty_phpdoc** [contrib]
```
Comment (keradus) [inline README.rst]: Symfony?

File: Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php [hunk 3]
```diff
@@ -36,6 +36,9 @@ public static function setUpBeforeClass()
     protected function getFixer()
     {
         $name = 'Symfony\CS\Fixer'.substr(get_called_class(), strlen(__NAMESPACE__), -strlen('Test'));
+        if (!class_exists($name)) {
+            $this->fail(sprintf('Fixer "%s" for test not found.', $name));
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php]: ???

File: Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php (hunk 1 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php]: I know, I don't like those in my projects.. do you want these?

File: Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php (hunk 3 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php]: DX, readability of the fail is better than a fatal stack dump

Comment (keradus) [inline Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php]: fail mark the test as failed, and here we are not in test...

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php]: Just throw an exaception?

File: Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php]: It's a user case thing. Taking things strictly - this is part of docblock, so it's not empty...
I don't like them too, but still...

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php]: Maybe we should define empty more loosly has having nothing useful in them?

File: README.rst (hunk 2 above)
Comment (SpacePossum) [inline README.rst]: I'll ask them if they reply on my PR overthere, till than I'm not going to the trouble to check it all out. If they want it we can always move it to SF level.

File: Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php (hunk 1 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php]: very well, let me try out some more regex stuff.

**Comment by SpacePossum** [comment]
closing till I figured out how to fix it (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1765#discussion_r52298636)

File: README.rst (hunk 2 above)
Comment (keradus) [inline README.rst]: No need to ask at first, I could send the PR (in the and you can treat PR as extra type of issue with code attached).
But seriously, I don't think there will be any violation for that rule.

The only case I met and why I want that fixer is our annotation cleaning fixers - eg when we remove `@access` annotation and there is nothing left in docblock

**Comment by keradus** [comment]
@SpacePossum , please play with utests so the utests will define everything that should be fixed and how - when it will be done (or it is already?) I will help with regexp

**Comment by SpacePossum** [comment]
Renamed, replaced the `fail` with `throw exception` (thanks @GrahamCampbell ) and added a bunch of integration tests (include case for removal of `@access` thanks @keradus).
Given that some of the tests are still running, I think the one thing (and most important thing) is to figure out the rule on which to remove a PHPDoc.
I like it as is, but I see the point @keradus is making.
It will be tricky to tweak though, so lets list some cases here.
I think we agree on these:
```php
/** */

/** 
*/

/** 
 *
 *
 */
```
However, the fixer also removes:
```php
/**
**/

/*******
 *
 *****/

/**
 * *
 */
```
I'm happy with both sets, so let me know were to aim for :)

**Comment by keradus** [comment]
For 6 cases you provided I woudn't touch only the last one, in which `*` is summary part of PHPDoc. Odd, not nice, but still valid.

**Comment by gharlan** [comment]
I like it as-is.

**Comment by keradus** [comment]
@SpacePossum All cases you provided are also in utests, am I right ?

**Comment by SpacePossum** [comment]
@keradus yes I think so, I checked and cannot think of anymore other cases, let me know I you have any in mind.

**Comment by keradus** [comment]
Cool!
Please, move it to Symfony level

**Comment by SpacePossum** [comment]
>Please, move it to Symfony level

:+1:  done :)

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**